### PR TITLE
Add required tags to account config

### PR DESCRIPTION
**Pull request by ThatJoeMoore**
Per the discussion in #332, this PR will allow us to add a list of required tags to an account config file. This list will then be enforced during `handel check` and `handel deploy`.

Note: This won't take effect immediately when merged. We'd have to update each account config file, then have all deployers download the new account config file.

Side note: should we explore keeping the account config files in a specially-named S3 bucket or Parameter Store value? Then we could roll out account config file changes more easily.

### CODE REVIEW

**Comment by ThatJoeMoore** [comment]
@dsw88 ready for review.

**Comment by ThatJoeMoore** [comment]
One sad thing - this PR ruins our beautiful test count of 500.  Now it's 501 😭

**Comment by dsw88** [comment]
Oh you're right! You'd better delete one of our tests before we merge so we can get back to 500. 😄 Haha jk, I'm reviewing now.

File: src/services/apiaccess/index.ts [hunk 1]
```diff
@@ -77,3 +77,5 @@ export const producedDeployOutputTypes = [
 ];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/apiaccess/index.ts]: I think this one should be set to false right?

File: src/services/cloudwatchevent/index.ts [hunk 2]
```diff
@@ -145,3 +153,5 @@ export const producedEventsSupportedServices = [
 export const producedDeployOutputTypes = [];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/cloudwatchevent/index.ts]: I think the Handel docs don't show that the CloudWatch Events service supports Tags (the rules don't, but we do tag the stack here), can we add that to the docs as part of this change?

File: src/services/iot/index.ts [hunk 3]
```diff
@@ -159,3 +167,5 @@ export const producedEventsSupportedServices = [
 export const producedDeployOutputTypes = [];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/iot/index.ts]: The Handel docs also don't show that Tags are supported, but we apparently do add tags to the stack. Can we update those docs as part of this PR?

File: src/services/kms/index.ts [hunk 4]
```diff
@@ -123,3 +124,5 @@ export const producedDeployOutputTypes = [
 ];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/kms/index.ts]: The Handel docs also don't show that Tags are supported for KMS, but we apparently do add tags to the stack. Can we update those docs as part of this PR?

File: src/services/ses/index.ts [hunk 5]
```diff
@@ -84,3 +84,5 @@ export const producedDeployOutputTypes = [
 ];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/ses/index.ts]: I think this one should be set to false right?

File: src/services/sns/index.ts [hunk 6]
```diff
@@ -182,3 +191,5 @@ export const producedDeployOutputTypes = [
 export const consumedDeployOutputTypes = [
     'cloudwatchevent'
 ];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/sns/index.ts]: The Handel docs also don't show that Tags are supported for KMS, but we apparently do add tags to the stack. Can we update those docs as part of this PR?

File: src/services/sqs/index.ts [hunk 7]
```diff
@@ -230,3 +238,5 @@ export const producedDeployOutputTypes = [
 ];
 
 export const consumedDeployOutputTypes = [];
+
+export const supportsTagging = true;
```
Comment (dsw88) [inline src/services/sqs/index.ts]: The Handel docs also don't show that Tags are supported for KMS, but we apparently do add tags to the stack. Can we update those docs as part of this PR?

**Comment by dsw88** [comment]
The PR looks great! Just a few comments, mostly about the fact that historically I haven't shown the "tags" parameter on things like SQS that don't support tags for the queue itself, but we are tagging the stacks at least and any of the resources that do support it. I'm thinking we should update the docs to show the tags element for those, what do you think?

**Comment by wiederhold** [comment]
> Per the discussion in #332, this PR will allow us to add a list of required tags to an account config file. This list will then be enforced during handel check and handel deploy.

Note: This won't take effect immediately when merged. We'd have to update each account config file, then have all deployers download the new account config file.

Side note: should we explore keeping the account config files in a specially-named S3 bucket or Parameter Store value? Then we could roll out account config file changes more easily.

☝️ Yes!!! I also REALLY like the idea of enforcement for handel configs at deploy. By putting them in a central location we can get to an ideal scenario where any organization (not just BYU) could enforce there own set of rules/guidelines at deployment for an entire stack. This is WAY better than AWS Config in its present state, and more to the point, gets the config rules closer to the dev pipeline than any other mechanism anyone's suggested. HUGE thumbs up... this is IMO brilliant!

**Comment by ThatJoeMoore** [comment]
Adding tags to the tagless services works for me.  We should probably note that, at this time, those resources don't support tagging, so we'll just be tagging the cloudformation.  But if AWS ever gets their act together, the tags should get automatically deployed the next time the CFN stack gets updated, so that could be really nice.

**Comment by ThatJoeMoore** [comment]
I think I'm going to go ahead and restructure some of our tagging documentation. Every service's tag schema is the same, so instead of documenting it in every single service, I'm going to have them all link to the main tagging document.  I think that'll make all of this much easier to keep clean and up-to-date.

**Comment by dsw88** [comment]
Great idea, yeah there's a lot of repeated text for tags on every service page, it would definitely be nice to consolidate it into a single location.

**Comment by ThatJoeMoore** [comment]
Alright, I think I've got all the docs nice and pretty.

**Comment by dsw88** [comment]
Awesome looks awesome, thanks for implementing this!

**Pull request merged**